File: cplus/leetcode/202606/algo3147.cpp
#include <bits/stdc++.h>
using namespace std;

// 七、其他线性 DP - 7.1 一维 DP
class Solution {
public:
    int maximumEnergy(vector<int>& energy, int k) {
        int ans = INT_MIN, n = energy.size();

        for (int i=0; i<n; i++) {
            if (i+k>=n) ans = max(energy[i], ans);
            else energy[i+k] = max(energy[i+k], energy[i+k] + energy[i]);
        }

        return ans;
    }
};

File: python/algo/202606/algo3147.py
from cmath import inf
from typing import List

# 七、其他线性 DP - 7.1 一维 DP
class Solution:
    def maximumEnergy(self, energy: List[int], k: int) -> int:
        ans, n = -inf, len(energy)
        
        for i in range(n):
            if i + k >= n: ans = max(ans, energy[i]) 
            else: energy[i+k] = max(energy[i+k], energy[i+k] + energy[i])
    
        return ans